### ISSUE TITLE
rbtray slows down java applications

### ISSUE THREAD

**Issue by lemon-juice** [benbuck/rbtray#33]
I have found out that rbtray has negative performance impact on DBeaver, which is a Java-based application for managing databases. When rbtray is running then certain tasks in DBeaver that render elements on screen take much more time to complete than when rbtray is not running. It is most noticeable when I am opening new tables in tabs by double-clicking on their names in the left navigation pane. Rbtray makes opening the tabs about twice as slow.

I've written about it in the [DBeaver issue tracker](https://github.com/dbeaver/dbeaver/issues/6821). That was about 4T Tray Minimizer but rbtray causes the same problem, probably some of the code is the same. My impression is that rbtray is slightly better with its performance impact than 4T Tray Minimizer but still noticeable.

I don't know if this problem affects only DBeaver or other Java applications, too. But this seems weird because opening new tabs in a Java UI should have nothing to do with minimizing whole windows so I don't think that rbtray should cause any slow downs - but it does.

I tested rbtray 4.10 on two different computers with Windows 10 64-bit and the same problem occurs on both.

**Comment by OndrejPetrzilka** [comment]
RBTray uses DLL injection and WH_MOUSE hook to handle mouse events in different process.
Maybe the Java app just generates a lot of events to handle...or has some performance issues related to executing code in injected DLL.

I've suggested #38 to provide a way around the issue.

**Comment by benbuck** [comment]
Hi @lemon-juice,

I just released version 4.11, which has a new feature from @OndrejPetrzilka to disable the RBTray mouse hook. Could you please try running with "--no-hook" on the command line to see if that helps with this issue?

Thanks,
Benbuck

**Comment by lemon-juice** [comment]
Thanks for the new release. The interesting thing is that the performance impact is overall much smaller when I test it now in the normal (mouse) mode. When opening tabs with table properties in dbeaver I don't notice any visible delays caused by rbtray, only when opening new empty sql tabs there seems to be a slight delay but it's not much longer than without rbtray running. But the --no-hook option does it's job - when I use it then there's no performance impact at all.

So overall things have improved for me with rbtray in normal mode but I'm not sure why. Meanwhile, I upgraded my computer, dbeaver has gone a long way in development (and is probably now based on a different Java version) and there have been a few Windows patches since I filed this bug here so there are many factors involved.
